### PR TITLE
[Refactor] refactor is_available, check_env

### PR DESCRIPTION
**Pull request by grimoire**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Refactor for `is_available`, `is_ops_available`, `check_env`, `get_version`...

## Modification

- move `is_available` into `BackendManager`, add `with_custom_ops` arguments.
- Add `check_env` into `BackendManager`, more detail env check.
- Add `get_version`.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdeploy/backend/base/backend_manager.py [hunk 1]
```diff
@@ -28,7 +28,50 @@ def build_wrapper(cls,
...
+                except NotImplementedError:
+                    backend_version = 'Unknown'
+            else:
+                backend_version = 'None'
+
+            info = f'{cls.backend_name}:\t{backend_version}'
+        except Exception:
+            info = f'{cls.backend_name}:\tCheckFailed'
+
+        log_callback(info)
```
Comment (lvhan028) [inline mmdeploy/backend/base/backend_manager.py:73]: `log_callback` is supposed to be verified not `None` before being accessed.

File: mmdeploy/backend/base/backend_manager.py [hunk 2]
```diff
@@ -89,3 +134,17 @@ def find(self, name: str) -> BaseBackendManager:
...
+def get_backend_manager(name: str) -> BaseBackendManager:
+    """Get backend manager.
+
+    Args:
+        name (str): name of the backend.
+    Returns:
+        BaseBackendManager: The backend manager of given name
+    """
+    from enum import Enum
+    if isinstance(name, Enum):
```
Comment (lvhan028) [inline mmdeploy/backend/base/backend_manager.py:148]: Is it necessary since the argument `name` is expected to be `str`?

Comment (grimoire) [reply mmdeploy/backend/base/backend_manager.py:148]: For backward compatible

**Review by lvhan028 — APPROVED** [review]

File: mmdeploy/apis/utils/calibration.py [hunk 3]
```diff
@@ -46,6 +43,10 @@ def create_calib_input_data(calib_file: str,
             'val', defaults to 'val'.
         device (str): Specifying the device to run on, defaults to 'cpu'.
     """
+    import h5py
+    import tqdm
```
Comment (AllentDan) [inline mmdeploy/apis/utils/calibration.py:47]: does it mean `tqdm` and `h5py` should be moved to optional.txt in requirements?

File: mmdeploy/backend/rknn/backend_manager.py [hunk 4]
```diff
@@ -38,6 +38,32 @@ def build_wrapper(cls,
...
+
+    @classmethod
+    def get_version(cls) -> str:
+        """Get the version of the backend."""
+        if not cls.is_available():
+            return 'None'
+        else:
+            import pkg_resources
+            try:
+                return pkg_resources.get_distribution('rknn').version
```
Comment (AllentDan) [inline mmdeploy/backend/rknn/backend_manager.py:63]: `rknn` package name should be `rknn-toolkit` or `rknn-toolkit2`

File: mmdeploy/apis/utils/calibration.py (hunk 3 above)
Comment (grimoire) [reply mmdeploy/apis/utils/calibration.py:47]: Reasonable.

**Review by AllentDan — APPROVED** [review]
LGTM

**Pull request merged**